src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -131,44 +119,26 @@ void executeInternal(SensorContext context) {
     }
   }
 
-  private static boolean requiresAnalyzerScannerExecution(String analyzerWorkDirPath) {
-    Path dir = analyzerOutputDir(analyzerWorkDirPath);
-    if (!Files.exists(dir)){
-      LOG.info("Analyzer working directory does not exist");
-      return true;
-    }
-
-    try ( DirectoryStream<Path> files = Files.newDirectoryStream(dir, p -> p.toAbsolutePath().toString().toLowerCase().endsWith(".pb"))) {
-      long count = StreamSupport.stream(files.spliterator(), false).count();
-      LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
-      return count == 0;
-    } catch (IOException e) {
-      LOG.warn("Could not check for .pb files in " + dir.toAbsolutePath().toString(), e);
-      return true;
-    }
-  }
-
   void analyze(boolean includeRules, SensorContext context) {
     if (includeRules) {
       LOG.warn("***********************************************************************************");
       LOG.warn("*                 Use MSBuild 14 to get the best analysis results                 *");
       LOG.warn("* The use of MSBuild 12 or the sonar-scanner to analyze C# projects is DEPRECATED *");
       LOG.warn("***********************************************************************************");
 
-      ImmutableMultimap<String, RuleKey> activeRoslynRulesByPartialRepoKey =
-        RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(context.activeRules()
-          .findAll()
-          .stream()
-          .map(ActiveRule::ruleKey)
-          .collect(toList()));
+      ImmutableMultimap<String, RuleKey> activeRoslynRulesByPartialRepoKey = RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(context.activeRules()
+        .findAll()
+        .stream()
+        .map(ActiveRule::ruleKey)
+        .collect(toList()));
 
       if (activeRoslynRulesByPartialRepoKey.keySet().size() > 1) {
         throw new IllegalArgumentException(
           "Custom and 3rd party Roslyn analyzers are only by MSBuild 14. Either use MSBuild 14, or disable the custom/3rd party Roslyn analyzers in your quality profile.");
       }
     }
 
-    String analysisSettings = analysisSettings(true, settings.getBoolean("sonar.cs.ignoreHeaderComments"), includeRules, context);
+    String analysisSettings = AnalysisInputXml.generate(true, settings.getBoolean("sonar.cs.ignoreHeaderComments"), includeRules, context,  CSharpSonarRulesDefinition.REPOSITORY_KEY, CSharpPlugin.LANGUAGE_KEY);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 210 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)